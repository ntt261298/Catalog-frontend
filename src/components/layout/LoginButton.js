import React from 'react';
import { connect } from 'react-redux';
import { PropTypes } from 'prop-types';
import { toggleLogin } from '../../actions/app';

const LoginButton = props => (
  <button type="button" onClick={props.toggleLogin} className="btn-login">Login</button>
);

LoginButton.propTypes = {
  toggleLogin: PropTypes.func.isRequired,
};

export default connect(null, { toggleLogin })(LoginButton);
