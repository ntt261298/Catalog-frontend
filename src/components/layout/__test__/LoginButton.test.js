import React from 'react';
import { shallow } from '../../../enzyme';
import { LoginButton } from '../LoginButton';

describe('Login button tests', () => {
  let wrapper;
  let props;
  let loginButton;
  const update = () => {
    loginButton = wrapper.find('button');
  };

  const setup = () => {
    wrapper = shallow(
      <LoginButton {...props} />,
    );
    update();
  };

  beforeEach(() => {
    props = {
      toggleLogin: jest.fn(),
    };
  });

  it('should render login button', () => {
    setup();
    // Expect the wrapper object to be defined
    expect(wrapper).toMatchSnapshot();
  });

  it('should call toggleLogin when login button is clicked', () => {
    setup();
    console.log(wrapper.debug());
    loginButton.simulate('click');
    expect(props.toggleLogin).toHaveBeenCalled();
  });
});
