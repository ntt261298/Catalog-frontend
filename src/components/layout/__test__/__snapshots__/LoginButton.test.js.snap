// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`src/components/layout/LoginButton.js should render login button 1`] = `
<button
  className="btn-login"
  onClick={[Function]}
  type="button"
>
  Login
</button>
`;
